R/fileDependency.R

```diff
@@ -0,0 +1,26 @@
+#' Create a file dependency that can be accessed by the client using javascript.
+#' @export
+fileDependency <- function(filename, version = '0.0.1'){
+  htmltools::htmlDependency(
+    name = basename(tools:::file_path_sans_ext(filename)),
+    version = version,
+    src = dirname(filename),
+    attachment = basename(filename)
+  )
+}
+
+#' Mark a string as an attachment
+#' @export
+attachment <- function(x){
+  structure(normalizePath(x), class = unique(c("ATTACHMENT", oldClass(x))))
```

Comment:
I just used the same code as used in `htmlwidgets:::JS`. I should admit I did not understand why `oldClass` was used there in the first place :smile: